src/lib/utils.ts

```diff
@@ -52,3 +52,15 @@ export const isToken = (assetReference: AssetReference | string) =>
 
 export const tokenOrUndefined = (assetReference: AssetReference | string) =>
   isToken(assetReference) ? assetReference : undefined
+
+// 0 is valid but falsy, dum language
+export const isValidAccountNumber = (
+  accountNumber: number | undefined | null,
+): accountNumber is number => {
+  if (accountNumber === undefined) return false
+  if (accountNumber === null) return false
+  if (Number.isNaN(accountNumber)) return false
+  if (!Number.isInteger(accountNumber)) return false
+  if (accountNumber >= 0) return true
+  return false
```

Comment:
feels like this could just be simplified to:
```
return Number.isInteger(accountNumber) && accountNumber >= 0
```

Reply:
not quite, but i did simplify it